### PR TITLE
Feature: Add goi18n v2 exporter

### PR DESCRIPTION
**Pull request by KuSh**
## Proposed changes

Add goi18n v2 exporter using new translate-toolkit capability
Also mark goi18n v1 format as supporting descriptions

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

I wasn't able to test exporting explanation from ui to files. I'll appreciate a little bit of help as I don't know weblate nor python too much

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ sentry_sdk>=1.6,<2
 siphashc>=1.2,<3.0
 social-auth-app-django==5.0.0
 social-auth-core==4.3.0
-translate-toolkit>=3.8.3,<3.9
+translate-toolkit>=3.8.6,<3.9
```
Comment (nijel) [inline requirements.txt:43]: The dependency bump is not really needed, the format will just not be supported if using an older version.

Comment (KuSh) [reply requirements.txt:43]: Ohh that's cool ! Thanks

Comment (nijel) [inline requirements.txt:43]:
```suggestion
translate-toolkit>=3.8.3,<3.9
```

**Comment by KuSh** [comment]
@nijel if you can give me advices on how to test "explanation" export into TranlsationUnit notes which should be exposed as "description" attribute in go-i18n formats, I'd love some help.

**Comment by nijel** [comment]
It should be shown as "Source string description", see https://hosted.weblate.org/translate/weblate/application/cs/?checksum=39b3998dcfe68cbf

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by nijel** [comment]
Description set in Weblate is not stored to the file, that one is kept in the database only.

**Pull request merged**